proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -445,6 +451,8 @@ lemma cancel_signal_corres:
           (invs' and st_tcb_at' ((=) (BlockedOnNotification ntfn)) t)
           (cancel_signal t ntfn)
           (cancelSignal t ntfn)"
+  apply (rule_tac Q="(\<lambda>s. sym_refs (state_refs_of' s))" in corres_cross_add_guard)
```

Comment:
It might be worth considering to alias this line or to make it into a method because we will be doing a lot in many places.

Reply:
Personally, I'd prefer to leave it as is, so that there's not another layer of definitions on top. But happy to do this if others would like to see it

Reply:
It'd be this line and the next. Might be indeed be worth a short method, because it's going to introduce a lot of duplication, and if we happen to tweak anything relevant it'll need a lot of updates.

Reply:
A method for both lines wouldn't apply in all cases that it's used; I add `sym_refs` in `blocked_cancel_ipc_corres`, and then only need the cross rule (as opposed to the other bits unpacking `invs'`)

Reply:
I've come up with a little method that is able to do both lines. Where should I put it? It needs to see `corres_cross_add_guard`, which is in Move, and also `state_refs_of_cross_eq`, which is in KHeap. It seems like KHeap doesn't import Move. Is that a mistake? If not, the earliest file I can see which sees both is TcbAcc. Should I put it there? If so, where in the file? Or is there a better place? I need to be able to use the method in Detype_R and IpcCancel_R, which seem to be in fairly different branches of the "tree"

Reply:
I think it would make sense if Move (or ArchMove) is imported by KHeap_R. I am not sure if there is a particular reason for that. Could you please try doing that? And, if it works, you can put your method just after `state_refs_of_cross_eq` I guess.